core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightBuilder.java

```diff
@@ -522,4 +525,33 @@ public String toString() {
             return name().toLowerCase(Locale.ROOT);
         }
     }
+
+    public static enum BoundaryScannerType implements Writeable {
+        SIMPLE, BREAK_ITERATOR;
```

Comment:
I don't think we _have_ to expose all options, was just thinking "if Lucene supports it ..". But having said that, from an API perspective this may be weird, so I'll take your proposal and add "boundary_scanner" with values "simple", "sentence" and "word". I'll do that.